realm/realm-library/src/main/java/io/realm/RealmResults.java

```diff
@@ -856,13 +866,24 @@ void notifyChangeListeners() {
             //FIXME: still waiting for Core to provide a fix
             //       for crash when calling _sync_if_needed on a cleared View.
             //       https://github.com/realm/realm-core/pull/1390
-            long version = table.sync();
+            long version;
+            try {
+                version = table.sync();
+            } catch (DeleteRealmListException e) {
+                // Although this RealmResults won't be updated anymore, it is good to give user a chance to do update.
+                version = TABLE_VIEW_VERSION_REALM_LIST_DELTED;
+                RealmLog.d("The parent RealmList has been deleted already.");
+            }
```

Comment:
I agree that the empty result is the "correct" behavior. Couldn't we just check for TABLE_VIEW_VERSION_REALM_LIST_DELTED in `isValid` ?